server/controllers/conversation.js

```diff
@@ -0,0 +1,51 @@
+const asyncHandler = require("express-async-handler");
+const Conversation = require("../models/Conversation");
+const Profile = require("../models/Profile");
+
+exports.createConversationViaProfile = asyncHandler(async (req, res, next) => {
+  const senderId = req.user.id;
+  const { profileId } = req.body;
+  if (!profileId) {
+    res.status(400);
+    throw new Error("Bad Request");
+  }
+  const receiver = await Profile.findById(profileId);
+  !receiver && res.status(400).json({ error });
+  const receiverId = receiver.userId;
+  const conversation = await Conversation.create({
+    members: [senderId, receiverId],
+  });
+  if (!conversation) {
+    res.status(500);
+    throw new Error("Something went wrong with the conversation");
+  }
+
+  res.status(201).json(conversation);
+});
+
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const { receiverId } = req.body;
+  const senderId = req.user.id;
+  !receiverId && res.status(400).json({ error });
+  const conversation = await Conversation.create({
+    members: [senderId, receiverId],
+  });
+  if (!conversation) {
+    res.status(500);
+    throw new Error("Something went wrong with the conversation");
+  }
+
+  res.status(201).json(conversation);
+});
+
+exports.getConversations = asyncHandler(async (req, res, next) => {
+  const userId = req.user.id;
+  const conversations = await Conversation.find({
```

Comment:
It would be nice to fetch the last message (along with it's read status) within each conversation :)

To fetch the last message within a conversation, you can either query the `Message` model or add an additional field to the `Conversation` model and save the last message's id within it (this will be done each time a new message is sent)

Eg for the second case:
```
const conversationSchema = new mongoose.Schema({
  members: {
    type: Array,
    ref: "User",
    required: true,
  },
lastMessage: {
   type: mongoose.Schema.Types.ObjectId,
   ref: "Message",
   required: true,
}
});
```

Reply:
Sounds good, I have added it

Reply:
@rajivtitus I was adding the lastMessage in GET Conversation and had a question, I am not sure if lastMessge should be required:true here as when i create the first conversation there might not be any message, Thoughts???

Reply:
@mayanksuchde That is correct, you can ignore the example case above as it was just to show you the idea 😄